proteuslib/unit_models/reverse_osmosis_1D.py

```diff
@@ -0,0 +1,800 @@
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+
+        # Units
+        units_meta = \
+            self.config.property_package.get_metadata().get_derived_units
+
+        nfe = self.config.finite_elements
+
+        # ==========================================================================
+        """ Unit model variables"""
+        self.A_comp = Var(
+            self.flowsheet().config.time,
+            self.solvent_list,
+            initialize=1e-12,
+            bounds=(1e-18, 1e-6),
+            domain=NonNegativeReals,
+            units=units_meta('length') * units_meta('pressure') ** -1 * units_meta('time') ** -1,
+            doc="""Solvent permeability coeff.""")
+        self.B_comp = Var(
+            self.flowsheet().config.time,
+            self.solute_list,
+            initialize=1e-8,
+            bounds=(1e-11, 1e-5),
+            domain=NonNegativeReals,
+            units=units_meta('length')*units_meta('time')**-1,
+            doc='Solute permeability coeff.')
+        # TODO: add water density to NaCl prop model and remove here (or use IDAES version)
+        self.dens_solvent = Param(
+            initialize=1000,
+            units=units_meta('mass')*units_meta('length')**-3,
+            doc='Pure water density')
+        self.recovery_vol_phase = Var(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            initialize=0.4,
+            bounds=(1e-2, 1 - 1e-6),
+            units=pyunits.dimensionless,
+            doc='Volumetric recovery rate')
+
+        def flux_mass_phase_comp_initialize(b, t, x, p, j):
+            if j in self.solvent_list:
+                return 5e-4
+            elif j in self.solute_list:
+                return 1e-6
+
+        def flux_mass_phase_comp_bounds(b, t, x, p, j):
+            if j in self.solvent_list:
+                ub = 3e-2
+                lb = 1e-4
+            elif j in self.solute_list:
+                ub = 1e-3
+                lb = 1e-8
+            return lb, ub
+
+        self.flux_mass_phase_comp = Var(
+            self.flowsheet().config.time,
+            self.feed_side.length_domain,
+            self.config.property_package.phase_list,
+            self.config.property_package.component_list,
+            initialize=flux_mass_phase_comp_initialize,
+            bounds=flux_mass_phase_comp_bounds,
+            units=units_meta('mass')*units_meta('length')**-2*units_meta('time')**-1,
+            doc='Mass flux across membrane')
+
+        self.area = Var(
+            initialize=10,
+            bounds=(1e-1, 1e3),
+            domain=NonNegativeReals,
+            units=units_meta('length')**2,
+            doc='Membrane area')
+
+        # TODO: Add initial value, bounds and real equation later. Just need for now since add_geometry() generates
+        #  cross-sectional area var that I am referencing
+        # self.feed_area_cross = Var(
```

Comment:
Is this code needed in the released version?

Reply:
Not until I add on additional variables and constraints for detailed calculations that account for concentration polarization (which will be in a subsequent PR).